src/Microsoft.OData.Core/UriParser/Parsers/ODataPathParser.cs

```diff
@@ -86,6 +87,20 @@ internal static void ExtractSegmentIdentifierAndParenthesisExpression(string seg
                 identifier = segmentText;
                 parenthesisExpression = null;
             }
+            else if (segmentText[segmentText.Length - 1] == ':')
+            {
+
+                if (segmentText.Length < 2 ||
+                    segmentText[segmentText.Length - 2] != ')')
+                {
+                    throw ExceptionUtil.CreateSyntaxError();
+                }
+
+
+                identifier = segmentText.Substring(0, parenthesisStart) + ":";
+                parenthesisExpression =
+                   segmentText.Substring(parenthesisStart + 1, segmentText.Length - identifier.Length - 2);

```

Comment:
Is the identifier used only for getting the length? if so it will be parenthesisStart+1 , right? which can directly be used. Or am I missing something?

Reply:
It is an output parameter as well.